src/coverlet.collector/DataCollection/CoverletCoverageCollector.cs

```diff
@@ -165,15 +165,28 @@ private void OnSessionEnd(object sender, SessionEndEventArgs e)
         /// <returns>Test modules list</returns>
         private IEnumerable<string> GetTestModules(SessionStartEventArgs sessionStartEventArgs)
         {
-            var testModules = sessionStartEventArgs.GetPropertyValue<IEnumerable<string>>(CoverletConstants.TestSourcesPropertyName);
-            if (_eqtTrace.IsInfoEnabled)
+            IEnumerable<string> testModules;
+
+            try
             {
-                _eqtTrace.Info("{0}: TestModules: '{1}'",
-                    CoverletConstants.DataCollectorName,
-                    string.Join(",", testModules ?? Enumerable.Empty<string>()));
+                testModules = GetTestSourcesValues(sessionStartEventArgs);
+                if (_eqtTrace.IsInfoEnabled)
+                {
+                    _eqtTrace.Info("{0}: TestModules: '{1}'",
+                        CoverletConstants.DataCollectorName,
+                        string.Join(",", testModules ?? Enumerable.Empty<string>()));
+                }
+            }
+            catch (MissingMethodException ex)
+            {
+                throw new Exception("Method not found, make sure to use .NET core SDK Version >= 2.2.300", ex);
```

Comment:
we should retrow same exception `MissingMethodException`

Reply:
I've now put the original exception message in the inner exception part and used our piece in the real exception message without concatinating it. I tried your formatting but it looked wired because the whole namespace down to the missing function was printed.

![image](https://user-images.githubusercontent.com/19378678/65747106-e33a2900-e100-11e9-8656-47df11b5498b.png)

With the way it is now it looks like this:

![image](https://user-images.githubusercontent.com/19378678/65747147-04027e80-e101-11e9-9c20-33332ded895a.png)

Reply:
Can you zoom on second image?I want to be sure to see the "real missing method" `GetPropertyValue`.

Reply:
Yes this screenshot should be better:

![image](https://user-images.githubusercontent.com/19378678/65840213-67b1c500-e316-11e9-9de5-8886a859d2bd.png)